### ISSUE TITLE
[BUG]: Creating a stack depends_on with extra parameters i.e. 'condition' fails

### ISSUE THREAD

**Issue by tcorzo** [azukaar/Cosmos-Server#295]
### What happened?

When trying to import a docker compose file with `services.{container}.depends_on.{other_container}.condition` the stack creation fails.
It's trying to unmarshal it as a string array :(

### What should have happened?

It should have correctly parsed the depends_on

### How to reproduce the bug?

This docker compose fails, removing the conditions results in circular dependecies and removing the dependencies altogether fails to deploy for infisical-related reasons:
```yaml
version: "3"

services:
  db-migration:
    container_name: infisical-db-migration
    depends_on:
      db:
        condition: service_healthy
    image: infisical/infisical:latest-postgres
    env_file: .env
    command: npm run migration:latest
    pull_policy: always
    networks:
      - infisical

  backend:
    container_name: infisical-backend
    restart: unless-stopped
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_started
      db-migration:
        condition: service_completed_successfully
    image: infisical/infisical:latest-postgres
    pull_policy: always
    env_file: .env
    ports:
      - 80:8080
    environment:
      - NODE_ENV=production
    networks:
      - infisical

  redis:
    image: redis
    container_name: infisical-dev-redis
    env_file: .env
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    ports:
      - 6379:6379
    networks:
      - infisical
    volumes:
      - redis_data:/data

  db:
    container_name: infisical-db
    image: postgres:14-alpine
    restart: always
    env_file: .env
    volumes:
      - pg_data:/var/lib/postgresql/data
    networks:
      - infisical
    healthcheck:
      test: "pg_isready --username=${POSTGRES_USER} && psql --username=${POSTGRES_USER} --list"
      interval: 5s
      timeout: 10s
      retries: 10

volumes:
  pg_data:
    driver: local
  redis_data:
    driver: local

networks:
  infisical:
```

### Relevant log output

```shell
Creating Service...                              
[OPERATION FAILED] Bad request: json: cannot unmarshal object into Go struct field ContainerCreateRequestContainer.services.depends_on of type []string%!(EXTRA int=400, string=DS003){"status":"error","message":"Bad request: json: cannot unmarshal object into Go struct field ContainerCreateRequestContainer.services.depends_on of type []string","code":"DS003"}
```

### System details

- OS: debian

**Comment by BearTS** [comment]
Can check this issue next week.

**Comment by azukaar** [comment]
Thanks Bear, althgouht Issue has been fixed already on the 0.16 branch, I added support for conditions
